packages/backend/src/apps/m365-excel/actions/create-table-row/index.ts

```diff
@@ -0,0 +1,174 @@
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'listFiles',
+          },
+        ],
+      },
+    },
+    {
+      key: 'tableId',
+      label: 'Table',
+      required: true,
+      description: 'Table to add new rows to',
+      type: 'dropdown' as const,
+      variables: false,
+      source: {
+        type: 'query' as const,
+        name: 'getDynamicData' as const,
+        arguments: [
+          {
+            name: 'key',
+            value: 'listTables',
+          },
+          {
+            name: 'parameters.fileId',
+            value: '{parameters.fileId}',
+          },
+        ],
+      },
+    },
+    {
+      label: 'Values',
+      key: 'columnValues',
+      type: 'multirow' as const,
+      required: true,
+      description: 'Specify values you want to insert in the new row',
+      subFields: [
+        {
+          key: 'index' as const,
+          type: 'dropdown' as const,
+          required: true,
+          variables: false,
+          placeholder: 'Column',
+          source: {
+            type: 'query' as const,
+            name: 'getDynamicData' as const,
+            arguments: [
+              {
+                name: 'key',
+                value: 'listTableColumns',
+              },
+              {
+                name: 'parameters.fileId',
+                value: '{parameters.fileId}',
+              },
+              {
+                name: 'parameters.tableId',
+                value: '{parameters.tableId}',
+              },
+            ],
+          },
+        },
+        {
+          key: 'value' as const,
+          type: 'string' as const,
+          required: true,
+          variables: true,
+          placeholder: 'Value',
+        },
+      ],
+    },
+  ],
+
+  async run($) {
+    const { fileId, tableId } = $.step.parameters
+    const columnValues = ($.step.parameters.columnValues as IJSONObject[]) ?? []
+    if (columnValues.length === 0) {
+      $.setActionItem({
+        raw: {
+          success: true,
+        },
+      })
+      return
+    }
+
+    const session = await WorkbookSession.acquire($, fileId as string)
+
+    const tableInfoResponse = await session.request<TableInfo>(
+      `/tables/${tableId}/range?$select=columnCount,rowIndex`,
+      'get',
+    )
+    const createRowResponse = await session.request<{ index: number }>(
+      `/tables/${tableId}/rows`,
+      'post',
+      {
+        data: {
+          index: null,
+          values: [
+            constructMsGraphArgment(
+              tableInfoResponse.data,
+              columnValues.map((val) => ({
+                index: Number(val.index),
+                value: String(val.value),
+              })),
+            ),
+          ],
+        },
+      },
+    )
+
+    $.setActionItem({
+      raw: {
+        relativeRowNumber: createRowResponse.data.index + 1,
```

Comment:
nit: is tableRowNumber a clearer way to represent it? I was confused initially while testing

Reply:
agreed, maybe sheetRowNumber and tableRowNumber?

Reply:
good call, done!